File: src/com/colacelli/irclib/connection/connectors/Connector.java
package com.colacelli.irclib.connection.connectors;

import com.colacelli.irclib.actors.User;
import com.colacelli.irclib.connection.Server;

import java.io.IOException;

public interface Connector {
    String ENTER = "\r\n";

    void connect(Server newServer, User newUser) throws IOException;

    void disconnect() throws IOException;

    String listen() throws IOException;

    void send(String text) throws IOException;
}

File: src/com/colacelli/irclib/connection/connectors/Connector.kt
package com.colacelli.irclib.connection.connectors

import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Server

import java.io.IOException

interface Connector {

    @Throws(IOException::class)
    fun connect(server: Server, user: User)

    @Throws(IOException::class)
    fun disconnect()

    @Throws(IOException::class)
    fun listen(): String?

    @Throws(IOException::class)
    fun send(text: String)

    companion object {
        const val ENTER = "\r\n"
    }
}
